src/client/utils/download/download.ts

```diff
@@ -91,7 +92,11 @@ function datasetToSeparatedValues(
         .map((value: any) => {
           let formatted: string;
           if (TimeRange.isTimeRange(value)) {
-            formatted = formatDateWithTimezone(value.start, essence.timezone);
+            if (value.end.valueOf() >= day.shift(value.start, essence.timezone, 1).valueOf()) {
```

Comment:
put this in a function `formatTimeRange` or sth. and use it in both places instead of duplicating the code